### PR TITLE
Workdir should be absolute path

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/dockerfile/checks/WorkdirIsAbsolute.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        name = "Ensure that WORKDIR values are absolute paths"
+        id = "CKV_DOCKER_10"
+        supported_instructions = ["WORKDIR"]
+        categories = [CheckCategories.CONVENTION]
+        super().__init__(name=name, id=id, categories=categories, supported_instructions=supported_instructions)
+
+    def scan_entity_conf(self, conf):
+        for mydir in conf:
+            mypath = mydir["value"]
+            if not re.match("(^/[A-z0-9-_+]*)|(^[A-z0-9-_+]:\\\\.*)|(^\\$[{}A-z0-9-_+].*)", mypath):
```
Comment (gruebel) [inline checkov/dockerfile/checks/WorkdirIsAbsolute.py:21]: please use `re.compile(...)` to create the regex as a "static" variable and then use it here, like here https://github.com/bridgecrewio/checkov/blob/f43ef11c3bc489cb1a28c24826ce4d778cf35813/checkov/arm/checks/resource/NSGRulePortAccessRestricted.py#L11

Comment (JamesWoolfenden) [reply checkov/dockerfile/checks/WorkdirIsAbsolute.py]: have done it. Whats does that achieve?

Comment (gruebel) [reply checkov/dockerfile/checks/WorkdirIsAbsolute.py]: then the `flake8` plugin `dlint` is able to check, if we have some issue inside the regex

**Review by gruebel — APPROVED** [review]
thanks 🚀

**Pull request merged**